packages/core-utils/src/query.js

```diff
@@ -163,13 +163,32 @@ function isParamApplicable(paramInfo, query, config) {
   return true;
 }
 
+/**
+ * Helper method which replaces OTP flex modes with single FLEX mode that's
+ * more useful and easier to work with.
+ */
+export function reduceOtpFlexModes(modes) {
+  return modes.reduce((prev, cur) => {
```

Comment:
It looks like [JavaScript's `Set`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Set) would fit well here (but still return an array).

Reply:
I'm a little confused as to how you see Set helping. The source array doesn't contain duplicates, the issue is taking `FLEX_DIRECT`, `FLEX_ACCESS`, and `FLEX_EGRESS` and combining those into the single `FLEX` item. If you have a cleaner solution I would love it!